### PR TITLE
DOC-8272-C4-Apple --Swift/ObjC N1QL Code Snippets

### PR DESCRIPTION
**Pull request by ibsoln**
https://issues.couchbase.com/browse/DOC-8272

DOC-9047-C4-Code

### CODE REVIEW

File: modules/objc/examples/code_snippets/SampleCodeTest.m [hunk 1]
```diff
@@ -1804,7 +1804,48 @@ - (void) dontTestQuerySyntaxJson {
...
 
+- (void) docsonly_QuerySyntaxN1QL() {
+    /* Documentation Only query-syntax-n1ql
+        Declared elsewhere: CBLDatabase* argDb
+    */
 
+    // tag::query-syntax-n1ql[]
+    NSError *error;
+
+    CBLDatabase *db == argDB;
```
Comment (jayahariv) [inline modules/objc/examples/code_snippets/SampleCodeTest.m:1813]: do you think this two variable declaration are important, and need to be displayed?

I mean, we could keep this above the `tag::query-syntax-n1ql[]`

File: modules/objc/examples/code_snippets/SampleCodeTest.m [hunk 2]
```diff
@@ -1804,7 +1804,48 @@ - (void) dontTestQuerySyntaxJson {
...
+
+    NSString *n1qlstr = "SELECT * FROM _ WHERE type = \"hotel\""; // <.>
```
Comment (jayahariv) [inline modules/objc/examples/code_snippets/SampleCodeTest.m:1816]: not sure, should we promote users to write N1QL query using `_` or with a valid database name?

File: modules/objc/examples/code_snippets/SampleCodeTest.m [hunk 3]
```diff
@@ -1804,7 +1804,48 @@ - (void) dontTestQuerySyntaxJson {
...
+
+- (void) docsonly_QuerySyntaxN1QLParams() {
+    /* Documentation Only query-syntax-n1ql-params
+        Declared elsewhere: CBLDatabase* argDB
+    */
+
+    // tag::query-syntax-n1ql-params[]
+    NSError *error;
+
+    CBLDatabase *db = argDB;
```
Comment (jayahariv) [inline modules/objc/examples/code_snippets/SampleCodeTest.m]: same, like above.

File: modules/objc/examples/code_snippets/SampleCodeTest.m [hunk 4]
```diff
@@ -1804,7 +1804,48 @@ - (void) dontTestQuerySyntaxJson {
...
+
+    NSString *n1qlstr = "SELECT * FROM _ WHERE type = $type"; // <.>
+
+    CBLQuery* thisQuery = [db createQuery: n1qlstr];
+
+    CBLQueryParameters* n1qlparams = [[CBLQueryParameters alloc] init];
+    [params setString: @"hotel" forName: @"type"]; // <.>
+
+    thisQuery.parameters = n1qlparams;
```
Comment (jayahariv) [inline modules/objc/examples/code_snippets/SampleCodeTest.m:1842]: don't think this is correct! 


But something like the below is a way to set a dynamic variable value to a N1QL query. 
```
NSString* typeVar = @"hotel";
NSString *n1qlstr = [NSString stringWithFormat: @"SELECT * FROM dbName WHERE type = %@", typeVar];
```

**Review by jayahariv — CHANGES_REQUESTED** [review]
please check the comments.

File: modules/objc/examples/code_snippets/SampleCodeTest.m (hunk 1 above)
Comment (ibsoln) [reply modules/objc/examples/code_snippets/SampleCodeTest.m:1813]: DONe

File: modules/objc/examples/code_snippets/SampleCodeTest.m (hunk 2 above)
Comment (ibsoln) [reply modules/objc/examples/code_snippets/SampleCodeTest.m:1816]: As far as I am aware Jay _ is fine to use; may change when collections and scopes are supported

File: modules/objc/examples/code_snippets/SampleCodeTest.m (hunk 3 above)
Comment (ibsoln) [reply modules/objc/examples/code_snippets/SampleCodeTest.m]: Done

File: modules/objc/examples/code_snippets/SampleCodeTest.m (hunk 4 above)
Comment (ibsoln) [reply modules/objc/examples/code_snippets/SampleCodeTest.m:1842]: Hm. Neat trick. Not sure Jay. I thought that we were supposed to use the Query parameters ... maybe both work, but which do we want people using? I'll follow this up and change asap.

Comment (jayahariv) [reply modules/objc/examples/code_snippets/SampleCodeTest.m:1842]: @ibsoln  I don't think this code will work, using QueryParams!

**Pull request merged**